### PR TITLE
DEV-2877 | Front end changes to add revenue program name to contributions table

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

This ticket/PR is a subtask of [DEV-1332 | Add Revenue Program to the contributions table display](https://news-revenue-hub.atlassian.net/browse/DEV-1332). 

The branch for this PR branches form [DEV-2633 | Add Revenue Program to serializer](https://news-revenue-hub.atlassian.net/browse/DEV-2633), which contains the server-side changes to support DEV-1332.

At time of writing, DEV-2633 is awaiting review and has not yet been merged into the "holding branch" (DEV-1332-add-revenue-program-to-the-contributions-table-display) that will have two subtasks merged into it and itself be merged into main. 

DEV-1332-add-revenue-program-to-the-contributions-table-display is the base for this PR.

Note that this PR contains server-side changes being reviewed separately, but that code does not need to be reviewed here, and once DEV-2633 has been merged into DEV-1332, that part of the diff will vanish.

#### What's this PR do?

This PR updates the donations page such that the donations table has an additional sortable column called "Revenue program" which corresponds to the name of a contribution's revenue program. 

As per AC in ticket, users with access to > 1 revenue program will see this column in the table, while those who do not will not.

#### Why are we doing this? How does it help us?

Make it easier for admins of multi-RP Orgs and staff members with access to multiple RPs to navigate contributions table in dashboard.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Log in to the review as an admin. Go to the contributions section. You should see a column header for 'Revenue program' and you should be able to sort the table by that column.

To test the other case, log in as a user who only has access to a single RP (for instance a user you create via account onboarding flow). Ensure there's at least one contribution to the RP otherwise there won't be table data. View the contributions table, and there should not be a column for revenue program.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

Overall, the contributions table in org dashboard could use improvements, probably on par with work Chris did in [DEV-489](https://news-revenue-hub.atlassian.net/browse/DEV-489), where he replaced react-table with MUI's table component. However, this is beyond scope of this PR.

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-2877 | Front end changes to add revenue program name to contributions table](https://news-revenue-hub.atlassian.net/browse/DEV-2877)
[DEV-2633 | Add Revenue Program to serializer](https://news-revenue-hub.atlassian.net/browse/DEV-2633)
[DEV-1332 | Add Revenue Program to the contributions table display](https://news-revenue-hub.atlassian.net/browse/DEV-1332)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No

### CODE REVIEW

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]
Small stuff (I hope).

File: spa/cypress/e2e/04-donations.cy.js [hunk 1]
```diff
@@ -379,3 +392,47 @@ describe('Donations list', () => {
     });
   });
 });
+
+describe.only('Table sorting for revenue program name', () => {
```
[the remark this answers is not in the document]
Comment (benjaminEwhite) [reply spa/cypress/e2e/04-donations.cy.js]: 008f2cb801

**Comment by benjaminEwhite** [comment]
I needed to delete the DEV-1332 branch (which is base for this PR) because unwanted changes had ended up there. That had side effect of closing this PR, so now reopening.

File: spa/src/components/donations/DonationsTable.js [hunk 2]
```diff
@@ -21,7 +21,12 @@ function DonationsTable({ columns = [], fetchDonations, pageIndex, refetch, onPa
   const [totalResults, setTotalResults] = useState(0);
 
   const convertOrderingToString = (ordering) => {
-    return ordering.map((item) => (item.desc ? `-${item.id}` : item.id)).toString();
+    return ordering
+      .map((item) => {
+        const normalizedId = item.id.replace('.', '__');
```
[the remark this answers is not in the document]
Comment (benjaminEwhite) [reply spa/src/components/donations/DonationsTable.js:29]: 3bb13769f

**Review by nrh-cklimas — APPROVED** [review]

**Review by x110dc — APPROVED** [review]

**Review by njharman — APPROVED** [review]
just looked at python, lgtm. not much there.

**Pull request merged**